### PR TITLE
Refeactored framework deployment utils

### PR DESCRIPTION
**Pull request by alejandrox1**
xref: #76206

Signed-off-by: Jorge Alarcon Ochoa <alarcj137@gmail.com>

**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
Refactor and cleanup e2e framework deployment utils
xref: #76206

**Special notes for your reviewer**:
This is intended to be the beginning of the refactoring of `framework/deployment_utils.go` into `framework/deployment`.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by cblecker** [comment]
/uncc
Don't have the bandwidth to review this at the moment.

/cc @andrewsykim 
Do you have specific reviewers helping with this effort, Andrew?

**Review by neolit123 — COMMENTED** [review]
LGTM @alejandrox1 
if you are ready with this we can remove the WIP.

first run `./hack/update-bazel.sh`
and include the resulted changes using `git add...`

then look at these failures:
```
test/e2e/apps/deployment.go:324:20: undefined: framework.UpdateDeploymentWithRetries
test/e2e/apps/deployment.go:466:20: undefined: framework.UpdateDeploymentWithRetries
test/e2e/apps/deployment.go:539:21: undefined: framework.UpdateDeploymentWithRetries
test/e2e/apps/deployment.go:685:22: undefined: framework.UpdateDeploymentWithRetries
test/e2e/apps/deployment.go:695:22: undefined: framework.UpdateDeploymentWithRetries
test/e2e/apps/deployment.go:706:22: undefined: framework.UpdateDeploymentWithRetries
test/e2e/apps/deployment.go:715:23: undefined: framework.UpdateDeploymentWithRetries
test/e2e/apps/deployment.go:722:23: undefined: framework.UpdateDeploymentWithRetries
test/e2e/apps/deployment.go:759:21: undefined: framework.UpdateDeploymentWithRetries
test/e2e/apps/deployment.go:865:20: undefined: framework.UpdateDeploymentWithRetries
test/e2e/apps/deployment.go:865:20: too many errors
# k8s.io/kubernetes/test/e2e/upgrades/apps
test/e2e/upgrades/apps/deployments.go:86:20: undefined: framework.UpdateDeploymentWithRetries
test/e2e/upgrades/apps/deployments.go:162:11: undefined: framework.UpdateDeploymentWithRetries
```
try building the suite locally with those commands i gave you on slack.
when done `git add` all fixes and force push your commit to the remote branch.

**Comment by neolit123** [comment]
/unassign timothysc
until WIP is done.

**Comment by alejandrox1** [comment]
Thank you so much for all of your comments @neolit123 I'm feeling a lot more comfortable getting into the framework.

I opened this pr based on the comments of #76800. Please tell me if it makes more sense to do multiple small PRs to refactor `framework/deployment_util.go` into `framework/deployment` or if it would be better for me to try and do it all at once. My thinking is that small changes are easier to review but it may not be such a big deal for others who are a lot more familiar with the code.

**Review by neolit123 — COMMENTED** [review]
thank you for the update @alejandrox1 

> I opened this pr based on the comments of #76800. Please tell me if it makes more sense to do multiple small PRs to refactor framework/deployment_util.go into framework/deployment or if it would be better for me to try and do it all at once. My thinking is that small changes are easier to review but it may not be such a big deal for others who are a lot more familiar with the code.

i think this current PR is a good example in terms of size vs contents.
PRs like this one are preferred instead of many lines of changes.

thanks
/priority backlog
/approve

**Comment by oomichi** [comment]
@alejandrox1 

Hello,

> I opened this pr based on the comments of #76800. Please tell me if it makes more sense to do multiple small PRs to refactor framework/deployment_util.go into framework/deployment or if it would be better for me to try and do it all at once. My thinking is that small changes are easier to review but it may not be such a big deal for others who are a lot more familiar with the code.

Does the above comment mean we will have more PRs which move all remaining code of ` test/e2e/framework/deployment_util.go` into a new path?
When I checked this PR at the first look, I was confused because this PR just moves one function.
It is better to mention about that on the PR message which mean we have more TODOs.

The code itself seems good for me anyways.

**Comment by alejandrox1** [comment]
Hi @oomichi 

> Does the above comment mean we will have more PRs which move all remaining code of ` test/e2e/framework/deployment_util.go` into a new path?
> When I checked this PR at the first look, I was confused because this PR just moves one function.
> It is better to mention about that on the PR message which mean we have more TODOs.

Thank you for your comment. I added some more context into the PR message.
I wanted to begin with one just one function to get some feedback on whether this is would be the optimal way of refactoring `framework/deployment_util.go` and I was also doubtful about the appropriate number of changes that I should commit in order to make reviewing easy.

File: test/e2e/framework/deployment/utils.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+import (
+	apps "k8s.io/api/apps/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	testutils "k8s.io/kubernetes/test/utils"
+)
+
+// UpdateDeploymentWithRetries updates the specified deployment with retries.
+func UpdateDeploymentWithRetries(c clientset.Interface, namespace, name string, applyUpdate testutils.UpdateDeploymentFunc) (*apps.Deployment, error) {
```
Comment (timothysc) [inline test/e2e/framework/deployment/utils.go]: what about all the other functions under deployment_util?

**Comment by timothysc** [comment]
meh, I'll approve and keep on trucking.  

However I'd prefer in the future to bundle shuffling together in complete units. 

/approve

**Pull request merged**